cloudify_suspend/workflows.py

```diff
@@ -11,31 +11,202 @@
 
 
-def _run_operation(ctx, operation, **kwargs):
-    full_operation_name = 'cloudify.interfaces.lifecycle.' + operation
-    ctx.logger.debug(operation)
-    graph = ctx.graph_mode()
+def _check_type(node, include_node_types, exclude_node_types):
+    """check that we have correct type for run action"""
+
+    # check by include types
+    if include_node_types:
+        for node_type in include_node_types:
+            if node_type not in node.type_hierarchy:
+                return False
+
+    # check by exclude types
+    if exclude_node_types:
+        for node_type in exclude_node_types:
+            if node_type in node.type_hierarchy:
+                return False
+
+    return True
+
+
+def _run_operation(ctx, graph, operation, **kwargs):
+    ctx.logger.debug("Run {}({})".format(operation, repr(kwargs)))
+
+    include_node_types = kwargs.get('include_node_types', [])
+    exclude_node_types = kwargs.get('exclude_node_types', [])
+    include_instances = kwargs.get('include_instances', [])
 
     for node in ctx.nodes:
-        if full_operation_name in node.operations:
+        if not _check_type(node, include_node_types, exclude_node_types):
+            continue
+
+        if operation in node.operations:
             for instance in node.instances:
+                # check for skip instances
+                if include_instances:
+                    if instance.id not in include_instances:
+                        continue
+                # add to run operation
                 sequence = graph.sequence()
                 sequence.add(
                     instance.send_event('Starting to {}'.format(operation)),
-                    instance.execute_operation(full_operation_name, **kwargs),
+                    instance.execute_operation(operation,
+                                               kwargs=kwargs),
                     instance.send_event('Done {}'.format(operation)))
 
-    return graph.execute()
-
 
 @workflow
 def suspend(ctx, **kwargs):
-    _run_operation(ctx, "suspend", **kwargs)
+    graph = ctx.graph_mode()
+    # deprecated
+    _run_operation(ctx, graph, 'cloudify.interfaces.lifecycle.suspend',
+                   **kwargs)
+    # new way
+    _run_operation(ctx, graph, 'cloudify.interfaces.freeze.suspend',
+                   **kwargs)
+    graph.execute()
 
 
 @workflow
 def resume(ctx, **kwargs):
-    _run_operation(ctx, "resume", **kwargs)
+    graph = ctx.graph_mode()
+    # new way
+    _run_operation(ctx, graph, "cloudify.interfaces.freeze.resume",
+                   **kwargs)
+    # deprecated
+    _run_operation(ctx, graph, "cloudify.interfaces.lifecycle.resume",
+                   **kwargs)
+    graph.execute()
+
+
+@workflow
+def statistics(ctx, **kwargs):
+    graph = ctx.graph_mode()
+    _run_operation(ctx, graph, "cloudify.interfaces.statistics.perfomance",
+                   **kwargs)
+    graph.execute()
+
+
+def _fs_prepare(ctx, graph, kwargs):
```

Comment:
@0lvin Can you add some documentation to this method , I think it will be helpful